tests/Integration/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCaseTest.php

```diff
@@ -292,33 +325,75 @@ public function testWhenNotificationIsForNonExistingDonation_newDonationIsCreate
 
 		$storeDonationCalls = $repositorySpy->getStoreDonationCalls();
 		$this->assertCount( 1, $storeDonationCalls, 'Donation is stored' );
-
+		$this->assertNull( $storeDonationCalls[0]->getId(), 'ID is not taken from request' );
 		$this->assertDonationIsCreatedWithNotficationRequestData( $storeDonationCalls[0] );
 	}
 
-	private function assertDonationIsCreatedWithNotficationRequestData( Donation $donation ) {
-		$this->assertEquals( 0, $donation->getPaymentIntervalInMonths(), 'Payment interval is always empty' );
+	public function testGivenRecurringPaymentForBookedDonation_newDonationIsCreated() {
+		$donation = ValidDonation::newBookedPayPalDonation();
+		$repositorySpy = new DonationRepositorySpy( $donation );
 
-		$this->assertNull( $donation->getId(), 'ID is not taken from request' );
+		$request = ValidPayPalNotificationRequest::newRecurringPayment( $donation->getId() );
+
+		$useCase = new HandlePayPalPaymentNotificationUseCase(
+			$repositorySpy,
+			new SucceedingDonationAuthorizer(),
+			$this->getMailer(),
+			new NullLogger()
+		);
+
+		$useCase->handleNotification( $request );
+
+		$this->assertCount( 1, $repositorySpy->getStoreDonationCalls() );
+		/** @var Donation $newDonation */
+		$newDonation = $repositorySpy->getStoreDonationCalls()[0];
```

Comment:
I'd probably have inlined this. Then you have it twice, but no local var, and no doc for it